addon/services/user-agent.js

```diff
@@ -1,106 +1,135 @@
-/* global FastBoot */
-
 import Service from '@ember/service';
-import { computed, get, observer, setProperties } from '@ember/object';
-import { readOnly } from '@ember/object/computed';
-import { assign } from '@ember/polyfills';
-import { isEqual } from '@ember/utils';
-import { assert } from '@ember/debug';
+import { computed, get, set } from '@ember/object';
 import { getOwner } from '@ember/application';
+import { deprecate } from '@ember/application/deprecations';
 import UAParser from 'ua-parser-js';
 
 export default Service.extend({
+  _parser: computed(() => new UAParser),
 
-  fastboot: computed(function() {
-    return getOwner(this).lookup('service:fastboot');
-  }),
+  parser: computed('extensions', {
+    get() {
+      deprecate(
+        'Usage of the private property `parser` is deprecated. If you need to fiddle around with this private property, please open an issue and we can discuss.',
+        false,
+        { id: 'ember-useragent.service.parser', until: '1.0.0' }
+      );
 
-  isFastBoot: readOnly('fastboot.isFastBoot'),
+      return get(this, '_parser');
+    }
+  }).readOnly(),
 
-  userAgent: computed(function() {
-    if (get(this, 'isFastBoot')) {
-      let headers = get(this, 'fastboot.request.headers');
-      let userAgent = headers.get('user-agent');
+  UAParser: computed(function () {
+    deprecate(
+      'Usage of the property `UAParser` is deprecated. To get the UAParser class, import it as `import UAParser from \'ua-parser-js\';`. To get the instance of that class used by this service, as this property previously incorrectly returned, get the `parser` property.',
```

Comment:
The current README makes it seem like `userAgent.UAParser` is the same thing as ` import UAParser from 'ua-parser-js';`, which it is not. The actually former is the instance of the latter used by the service.

Since this is "broken" anyway and I believe that these private implementation details should be fiddled around with, I'd like to deprecate this.

Reply:
This is a really good point, thanks!